src/components/widgets/Toggle.tsx

```diff
@@ -0,0 +1,247 @@
+import { CSSProperties } from "react";
+import { Subset } from "../../definitions/types";
+
+import { ThemeRole } from "../theming/types";
+import { useMemo, useState } from "react";
+
+// Definitions
+import { primaryFont } from "../../styleDefinitions/typography";
+import { blue } from "@material-ui/core/colors";
+import { gray } from "../../definitions/colors";
+import { useUITheme } from "../theming";
+import { merge, uniqueId } from "lodash";
+
+// Type definitions for working with styles.
+type ToggleColorStyleSpec = {
+  backgroundColor: CSSProperties["color"];
+  labelColor: CSSProperties["color"];
+  knobColor: CSSProperties["color"];
+  borderColor: CSSProperties["color"];
```

Comment:
One current drawback of the style implementation is that the smart calculations only happen if there's a theme defined. If there isn't, the user has to provide these specific style overrides to customize the colors. Should we incorporate a "main hue" prop to the style ovveride?

Reply:
good idea. Yeah i think i like a "main hue" prop instead of the different border and background colors. If we end up needing that sort of control later we can add it, but i'd expect for the vast majority of cases it'll be easier for the user who's not using a theme to pass in just one color and have it all work nicely.